tests/unit/jobs/test_views.py

```diff
@@ -2367,3 +2367,168 @@ def test_update_submission_for_malformed_metrics(self):
+
+    def test_update_leaderboard_when_data_for_missing_and_invalid_metrics(self):
+        self.url = reverse_lazy(
+            "jobs:update_leaderboard_data",
+            kwargs={"leaderboard_data_pk": self.leaderboard_data.pk},
+        )
+        self.data = {
+            "leaderboard_data": '{"metric1": "123", "metric3": "456"}'
+        }
+        expected = {
+            "error": "Following metrics ['metric2'] are missing and following metrics are invalid ['metric3'] in the leaderboard data"
+        }
+        self.client.force_authenticate(user=self.challenge_host.user)
+        response = self.client.patch(self.url, self.data)
+        self.assertEqual(response.data, expected)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_update_leaderboard_when_data_for_missing_metrics(self):
+        self.url = reverse_lazy(
+            "jobs:update_leaderboard_data",
+            kwargs={"leaderboard_data_pk": self.leaderboard_data.pk},
+        )
+        self.data = {
+            "leaderboard_data": '{"metric1": "123"}'
+        }
+        expected = {
+            "error": "Following metrics are missing in the leaderboard data: ['metric2']"
+        }
+        self.client.force_authenticate(user=self.challenge_host.user)
+        response = self.client.patch(self.url, self.data)
+        self.assertEqual(response.data, expected)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_update_leaderboard_when_data_for_extra_metrics(self):
+        self.url = reverse_lazy(
+            "jobs:update_leaderboard_data",
+            kwargs={"leaderboard_data_pk": self.leaderboard_data.pk},
+        )
+        self.data = {
+            "leaderboard_data": '{"metric1": 123, "metric2": 123, "metric3": 123}'
+        }
+        expected = {
+            "error": "Following metrics are invalid in the leaderboard data: ['metric3']"
+        }
+        self.client.force_authenticate(user=self.challenge_host.user)
+        response = self.client.patch(self.url, self.data)
+        self.assertEqual(response.data, expected)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_update_leaderboard_when_data_for_malformed_metrics(self):
+        self.url = reverse_lazy(
+            "jobs:update_leaderboard_data",
+            kwargs={"leaderboard_data_pk": self.leaderboard_data.pk},
+        )
+        self.data = {
+            "leaderboard_data": '{"metric1": "123", "metric2": "123"}'
+        }
+        expected = {
+            "error": "Values for following metrics are not of float/int: [('metric1', <class 'str'>), ('metric2', <class 'str'>)]"
+        }
+        self.client.force_authenticate(user=self.challenge_host.user)
+        response = self.client.patch(self.url, self.data)
+        self.assertEqual(response.data, expected)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```

Comment:
I think you missed a case when leaderboard data is successfully updated.

Reply:
Can I work on this?

Reply:
@Ayukha Can you please add this case as well.